### ISSUE TITLE
Zillions of errors when used on incorrect architecture

### ISSUE THREAD

**Issue by bperrybap** [adafruit/TinyWireM#8]
If a user includes TinyWireM.h in a sketch and builds for say an Uno board, they will get tons of errors.
They can easily be confused and not recognize the issue. (I have seen this first hand)
The current library.properties says the library works on any architecture.
The properties files should indicate that it is AVR only.
But even specifying that the library architecture is AVR only does not fully solve the issue.
So to fully remedy the issue, TinyWireM.h should also use conditionals to explicitly check the processor type being used and crash the compile with a meaningful error message when attempting to use it on an unsupported processor.

Even better would be to work with the Attiny core maintainer and get this library into to actual core so that it is only available when that tiny core type is being used.

**Comment by ladyada** [comment]
yes, it has been added to the adafruit arduino boards package as Wire, so you can just use wire :)

**Comment by bperrybap** [comment]
Nice. But I see that the Wire library in the adafruit board package includes the TinyWireM code which also suffers from the same bug reported in issue #7 
I'll go file a bug over there for it.
